src/cmd-run

```diff
@@ -14,18 +14,20 @@ dn=$(dirname "$0")
 VM_DISK=
 VM_MEMORY=2048
 VM_PERSIST=0
+VM_PERSIST_IMG=
 VM_NCPUS="${VM_NCPUS:-$(nproc)}"
 VM_SRV_MNT=
 SSH_PORT=${SSH_PORT:-}
 USAGE="Usage: $0 [-d /path/to/disk.qcow2] [--] [qemu options...]
 Options:
-    -d DISK     Root disk drive (won't be changed by default)
-    --persist   Don't create a temporary snapshot
-    -i FILE     File containing an Ignition config
-    --srv src   Mount (via 9p) src on the host as /var/srv in guest
-    -m MB       RAM size in MB (2048)
-    -p PORT     The port on localhost to map to the VM's sshd. [2222]
-    -h          this ;-)
+    -d DISK               Root disk drive (won't be changed by default)
+    --persist             Don't create a temporary snapshot
```

Comment:
I think we should never overwrite something in `builds/`; i.e. when `-d` is not provided.  How about in that case we instead make a qcow2 overlay in `tmp/${imagename}` ?

Reply:
> I think we should never overwrite something in `builds/`

agree

Reply:
I agree, though... to confirm this isn't something that this patch introduces, right?

Reply:
Oh, true.  I hadn't noticed `--persist` existed before.

I guess then my argument would change to if we're *promoting* this tooling we should have it DTRT.

Reply:
> Oh, true. I hadn't noticed `--persist` existed before.


Yeah I didn't know about it but was identified in some of the multi-arch work. I even asked in https://github.com/coreos/coreos-assembler/pull/408#issuecomment-475056628 if anyone was using it. Maybe we should take a look at what jakub is doing and see if we can find a clearer path forward.

Reply:
Hmm, so it looks like `--persist` dates back to the original form as in https://github.com/cgwalters/playground/blob/master/coreos-run-qemu. Though offhand I don't think we should have this behaviour built-in. The only time I'd want to use it would be to target a disk which has a backing file, but that's exactly what this patch is adding. How about:

1. we drop `--persist` as it is
2. we rename `--persist-to-img IMG` in this patch to `--persist IMG`
3. without `--persist IMG`, we default to creating an image somewhere and passing its fd to qemu as described in https://github.com/coreos/coreos-assembler/pull/408#issuecomment-485830675. This ensures that `Ctrl-A S` becomes harmless. (My tmux prefix being `Ctrl-A`, I've definitely accidentally done this :| ).

Reply:
Basically never use `-snapshot` and just let the the "fronting" image be permanent or not based on `--persist IMG`.